This file is part of MXE. See LICENSE.md for licensing information.

Contains ad hoc patches for cross building.

Edit:There was a mess in OpenAL' `CMakeLists.txt` - `CHECK_INCLUDE_FILES` has been misused.

From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Guilherme Lima Bernal <dev@lbguilherme.com>
Edited by: Patryk (PsychoX) L. <psychoxivi+mxe@gmail.com>
Date: 2018-08-23 13:12:36.274477600 +0200
Subject: [PATCH 1/1] fixes for cross-building


diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1111111..2222222 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -965,7 +965,8 @@
 OPTION(ALSOFT_REQUIRE_MMDEVAPI "Require MMDevApi backend" OFF)
 IF(HAVE_WINDOWS_H)
     # Check MMSystem backend
-    CHECK_INCLUDE_FILES("windows.h;mmsystem.h" HAVE_MMSYSTEM_H -D_WIN32_WINNT=0x0502)
+    CHECK_INCLUDE_FILES("windows.h;mmsystem.h" HAVE_MMSYSTEM_H)
+       ADD_DEFINITIONS("-D_WIN32_WINNT=0x0502")
     IF(HAVE_MMSYSTEM_H)
         CHECK_SHARED_FUNCTION_EXISTS(waveOutOpen "windows.h;mmsystem.h" winmm "" HAVE_LIBWINMM)
         IF(HAVE_LIBWINMM)
@@ -975,6 +976,7 @@
                 SET(BACKENDS  "${BACKENDS} WinMM,")
                 SET(ALC_OBJS  ${ALC_OBJS} Alc/backends/winmm.c)
                 SET(EXTRA_LIBS winmm ${EXTRA_LIBS})
+                SET(PKG_CONFIG_LIBS "${PKG_CONFIG_LIBS} -lwinmm")
             ENDIF()
         ENDIF()
     ENDIF()
@@ -988,6 +990,7 @@
             SET(BACKENDS  "${BACKENDS} DirectSound${IS_LINKED},")
             SET(ALC_OBJS  ${ALC_OBJS} Alc/backends/dsound.c)
             ADD_BACKEND_LIBS(${DSOUND_LIBRARIES})
+            SET(PKG_CONFIG_LIBS "${PKG_CONFIG_LIBS} -luuid -lole32")
         ENDIF()
     ENDIF()

@@ -999,6 +1002,7 @@
             SET(HAVE_MMDEVAPI 1)
             SET(BACKENDS  "${BACKENDS} MMDevApi,")
             SET(ALC_OBJS  ${ALC_OBJS} Alc/backends/mmdevapi.c)
+            SET(PKG_CONFIG_LIBS "${PKG_CONFIG_LIBS} -lole32")
         ENDIF()
     ENDIF()
 ENDIF()
